3_javascript/javascript1/week4/Homework/script.js

```diff
@@ -0,0 +1,188 @@
+const hello = [
+  "Hello human! what's up?",
+  "hi, how are you doing?",
+  "What's up?",
+  "Ahoy matey! How are ye?",
+  "What's shaking?",
+];
+
+// Store voices
+let voices = [];
+// UI elements
+const startBtn = document.getElementById("startBtn");
+const result = document.getElementById("result");
+const processing = document.getElementById("processing");
+
+// Wait on voices to be loaded before fetching list
+window.speechSynthesis.onvoiceschanged = () => {
+  voices = window.speechSynthesis.getVoices();
+};
+
+// Speech to Text
+const SpeechRecognition =
+  window.SpeechRecognition || window.webkitSpeechRecognition;
+let toggleBtn = null;
+if (typeof SpeechRecognition === "undefined") {
+  startBtn.remove();
+  result.innerHTML =
+    "<strong>Your browser does not support Speech API. Please download latest Chrome version.<strong>";
+} else {
+  const recognition = new SpeechRecognition();
+  recognition.continuous = true; // If true, the silence period is longer
+  recognition.interimResults = true;
+  recognition.onresult = (event) => {
+    const current = event.resultIndex;
+    const recognitionResult = event.results[current];
+    const recognitionText = recognitionResult[0].transcript;
+
+    if (recognitionResult.isFinal) {
+      processing.innerHTML = "processing ...";
+
+      const reply = getReply(recognitionText);
+      const p = document.createElement("p");
+      p.innerHTML = `<strong>You said:</strong> ${recognitionText} </br><strong>Benjamin said:</strong> ${reply}`;
+      processing.innerHTML = "";
+      result.appendChild(p);
+
+      readOutLoud(reply);
+    } else {
+      processing.innerHTML = `listening: ${recognitionText}`;
+    }
+  };
+  let listening = false;
+  toggleBtn = () => {
+    if (listening) {
+      recognition.stop();
+      startBtn.textContent = "Start listening";
+    } else {
+      recognition.start();
+      startBtn.textContent = "Stop listening";
+    }
+    listening = !listening;
+  };
+  startBtn.addEventListener("click", toggleBtn);
+}
+
+// Processor
+function getReply(rawText) {
+  let text = rawText.replace(/\s/g, "").replace(/\'/g, "");
```

Comment:
why are you removing the blank spaces? Because of this a lot of conditions won't be triggered e.g 

`rawText: 'Remove fishing from my todo'`
`text: 'Removefishingfrommytodo' `

` text.includes("remove fishing from my todo") `never will be triggered